pandas/core/dtypes/dtypes.py

```diff
@@ -1051,35 +1055,69 @@ class IntervalDtype(PandasExtensionDtype):
     num = 103
     _metadata = (
         "subtype",
-        "closed",
+        "inclusive",
     )
     _match = re.compile(
-        r"(I|i)nterval\[(?P<subtype>[^,]+)(, (?P<closed>(right|left|both|neither)))?\]"
+        r"(I|i)nterval\[(?P<subtype>[^,]+)(, ("
+        r"?P<inclusive>(right|left|both|neither)))?\]"
```

Comment:
i think you need to accept closed here (as wel as inclusive) for a string dtype to get the warning. Is this tested?

Reply:
Hi @jreback 
1)
For Regex Pattern written, to create IntervalDtype object using string dtype 
```
pd.IntervalDtype(subtype='interval[int64, right]')
```
```
interval[int64, right]
```
The regex will assign subtype='int64' and inclusive='right'
Thus, I think it is not required to update regex to accept  closed here (as well as inclusive) for a string dtype.
2) Currently, there is not this test. Will add this test.